department-of-veterans-affairs/abd-vro
# MCP-2580 Offramp when VRO cant retrieve MAS annotations

## What was the problem?
User Story

As a VRO stakeholder, I want to have confidence that when claims cannot complete processing in VRO due to errors, the claims are appropriately offramped so that they will be processed manually by claim adjudicators.

Error Scenario

VRO cannot successfully retrieve annotation data via the MAS endpoint

Logs should record offramp reason: annotationDataRequestFailed

Associated tickets or Slack threads:

- [MCP-2580](https://amida.atlassian.net/browse/MCP-2580)

## How does this fix it?[^1]
We introduced error handling into the multicast of Camel for when there is no response from MAS annotations.  If there is no response from MAS annotations, we offramp the claim with the message "annotationDataRequestFailed", and do NOT continue the processing with getting LH data etc.

## How to test this PR
- Bring up containers
- test automatedClaim endpoint with collection id: 369 and benefit claim id: 1369
- verify in app container logs that we offramp the claim after not receiving MAS annotations


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
